airflow/upgrade/rules/custom_executors_require_full_path_rule.py

```diff
@@ -0,0 +1,40 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.configuration import conf
+from airflow.upgrade.rules.base_rule import BaseRule
+
+
+class CustomExecutorsRequireFullPathRule(BaseRule):
+    """
+    CustomExecutorsRequireFullPathRule class to ease upgrade to Airflow 2.0
+    """
+    title = "Custom Executors now require full path"
+    description = """\
+In Airflow-2.0, loading custom executors via plugins is no longer required.
+To load a custom executor, you have to provide a full path to the the custom executor module.
+                  """
+
+    def check(self):
+        executor = conf.get(section="core", key="executor")
+        if executor.count(".") <= 1:
```

Comment:
I think this is not a good solution because the custom executor can be in the main module and then it will only have one dot. This is a common practice for some teams to keep an eye on what is exported out. I think you can try to load this module here using import_string. If we succeed, we have success. If it failed, this value must be checked.

Reply:
Ok. noted!

Reply:
Also, I think this will not catch the case where a user has imported an Executor via Pluings but no set it in `[core] executor`.

I think a better solution would be to get the executor imported via plugin list: 

```python
from airflow.plugins_manager import executors_modules
if executors_modules:    
    print(executors_modules)
```

This should give you that list that should not be imported

Reply:
if the executor is not used it does not need to be updated in the config file. This rule checks that the configuration is valid 

plugin checking is a bit of a problem as it makes it difficult to write a plugin that is compatible with Airflow 2.0 and Airflow 1.10 at the same time.

Reply:
The check name is `CustomExecutorsRequireFullPathRule` so we should check if Custom Exercutors are used.

And in this case specifically, it is just a module, so the only thing a user needs to do is instead of using Airflow Plugin for Executor, they should use normal Python module / Python packages explained in our module management guide in Airflow docs